addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -143,56 +143,74 @@ def _update_accrual(self):
         """
         today = fields.Date.from_string(fields.Date.today())
 
+
         holidays = self.search([('accrual', '=', True), ('state', '=', 'validate'),
                                 '|', ('date_to', '=', False), ('date_to', '>', fields.Datetime.now()),
                                 '|', ('nextcall', '=', False), ('nextcall', '<=', today)])
 
         for holiday in holidays:
-            values = {}
-
-            delta = relativedelta(days=0)
-
-            if holiday.interval_unit == 'weeks':
-                delta = relativedelta(weeks=holiday.interval_number)
-            if holiday.interval_unit == 'months':
-                delta = relativedelta(months=holiday.interval_number)
-            if holiday.interval_unit == 'years':
-                delta = relativedelta(years=holiday.interval_number)
-
-            values['nextcall'] = (holiday.nextcall if holiday.nextcall else today) + delta
-
-            period_start = datetime.combine(today, time(0, 0, 0)) - delta
-            period_end = datetime.combine(today, time(0, 0, 0))
-
-            # We have to check when the employee has been created
-            # in order to not allocate him/her too much leaves
-            creation_date = fields.Datetime.from_string(holiday.employee_id.create_date)
+            employee_ids = {"company": holiday.mode_company_id.user_ids,
+                            "department": holiday.department_id.member_ids,
+                            "category": holiday.category_id.employee_ids,
+                            "employee": [holiday.employee_id]
```

Comment:
No need to create a list. We can loop on a recordset with one element, no problem.